Packs/AWS-Route53/Integrations/AWSRoute53/AWSRoute53.py

```diff
@@ -320,47 +268,69 @@ def test_dns_answer(args):
             'Protocol': response['Protocol']
         })
 
-        ec = {'AWS.Route53.TestDNSAnswer': response}
-        return create_entry('AWS Route53 Test DNS Answer', data, ec)
+        return create_entry('AWS Route53 Test DNS Answer', data, response, 'AWS.Route53.TestDNSAnswer')
 
     except Exception as error:
-        return error
+        return CommandResults(content_format=EntryFormat.TEXT, entry_type=EntryType.ERROR, readable_output=str(error))
+
+
+def main():  # pragma: no cover
+    params = demisto.params()
+    command = demisto.command()
+    aws_default_region = params.get('defaultRegion')
+    aws_role_arn = params.get('roleArn')
+    aws_role_session_name = params.get('roleSessionName')
+    aws_role_session_duration = params.get('sessionDuration')
+    aws_role_policy = None
+    aws_access_key_id = params.get('credentials', {}).get('identifier') or params.get('access_key')
+    aws_secret_access_key = params.get('credentials', {}).get('password') or params.get('secret_key')
+    verify_certificate = not params.get('insecure', True)
+    timeout = params.get('timeout')
+    retries = params.get('retries', DEFAULT_RETRIES)
 
-
-def test_function():
     try:
-        client = aws_session()
-        response = client.list_hosted_zones()
-        if response['ResponseMetadata']['HTTPStatusCode'] == 200:
-            return 'ok'
+        validate_params(aws_default_region, aws_role_arn, aws_role_session_name, aws_access_key_id,  # noqa
+                        aws_secret_access_key)
 
-    except Exception as error:
-        return error
+        aws_client = AWSClient(aws_default_region, aws_role_arn, aws_role_session_name,  # noqa
+                               aws_role_session_duration, aws_role_policy, aws_access_key_id, aws_secret_access_key,
+                               verify_certificate, timeout, retries)
+
+        args = demisto.args()
 
+        demisto.info(f'Command being called is {demisto.command()}')
+        if command == 'test-module':
+            client = aws_client.aws_session(service=SERVICE)
```

Comment:
Refactor with Extract Method.
In the future we might want to test the module differently, or add tests, and also unit-test that method.